webapp/src/Service/ExternalContestSourceService.php

```diff
@@ -1444,6 +1448,10 @@ protected function importSubmission(Event $event, EventData $data): void
                 $zipUrl = $data->files[0]->href;
                 if (preg_match('/^https?:\/\//', $zipUrl) === 0) {
                     // Relative URL, prepend the base URL.
+                    // If both the base path ends with a / and the zip URL starts with one, drop one of them
+                    if (str_ends_with($this->basePath, '/') && str_starts_with($zipUrl, '/')) {
+                        $zipUrl = substr($zipUrl, 1);
+                    }
```

Comment:
I think this is wrong (and could maybe be specified a bit more clearly in the spec). If `$zipUrl` starts with a `/`, then it should be appended after the host part of `$this->basePath` per specification of how to resolve relative URIs.

I think it would be better here to use a library or so, since simply stripping a `/` and then concatenating is not correct. Also, I wonder whether the prefix `/` is intended behaviour by PC^2.

Reply:
I think pc2 is wrong currently since they don’t support multi contest in the api. They are working on this, so maybe we should indeed not commit this until then

Reply:
So also revert https://github.com/DOMjudge/domjudge/pull/2371/commits/6680f111c01276d2bb9c7861b30a7eb792e26df4 ?

Reply:
I'd say so, yes.

Besides that, the existing implementation doesn't handle a relative URL that starts with a `/` correctly, but that's a separate issue.